### PR TITLE
fix: loader stomping context from other loaders

### PR DESCRIPTION
**Pull request by JakobJingleheimer**
Fixes https://github.com/testdouble/testdouble.js/issues/525
Fixes https://github.com/testdouble/testdouble.js/issues/528

### CODE REVIEW

File: lib/quibble.mjs [hunk 1]
```diff
@@ -55,15 +55,15 @@ export async function resolve (specifier, context, nextResolve) {
...
 
     const quibbledUrl = addQueryToUrl(url, '__quibble', stubModuleGeneration)
 
     if (url.startsWith('node:') && !getStubsInfo(quibbledUrl)) {
-      return { url }
+      return { ...ctx, url } // It's allowed to change ctx for a builtin (but unlikely)
     }
 
-    return { url: quibbledUrl }
+    return { ...ctx, url: quibbledUrl }
```
Comment (JakobJingleheimer) [inline lib/quibble.mjs:66]: Not sure if I should DRY this up?

```js
const resolvedUrl = (url.startsWith('node:') && !getStubsInfo(quibbledUrl))
  ? url
  : quibbledUrl;

return { ...ctx, url: resolvedUrl }
```

**Comment by JakobJingleheimer** [comment]
Sorry, for the noise: I tried to rename the branch to something more useful. I swear I've done that before without GitHub flipping the table.

**Comment by giltayar** [comment]
I'm missing a test that checks this, but I'll try and add one now.

**Comment by JakobJingleheimer** [comment]
You're welcome to use the tests from jakobjingleheimer/nodejs-loaders. Those loaders all work with testdouble. I can add some here if you'd like (they use the same node:assert & node:test)

**Comment by searls** [comment]
Landed in v0.9.2

**Pull request merged**